### PR TITLE
Fix failing test on R-devel

### LINKED ISSUE #2527: CRAN failures on R-devel
```
 ══ Failed tests ════════════════════════════════════════════════════════════════
    ── Failure ('test-get_source_expressions.R:97:5'): Multi-byte character truncated by parser is ignored ──
    error$message (`actual`) not identical to "unexpected input" (`expected`).
    
    `actual`:   "unexpected invalid token"
    `expected`: "unexpected input"        
    Backtrace:
        ▆
     1. └─lintr (local) with_content_to_parse(...) at test-get_source_expressions.R:96:3
     2.   └─base::eval(substitute(code), envir = content_env) at test-get_source_expressions.R:12:3
     3.     └─base::eval(substitute(code), envir = content_env)
     4.       └─testthat::expect_identical(error$message, "unexpected input") at test-get_source_expressions.R:97:5
    
    [ FAIL 1 | WARN 0 | SKIP 1 | PASS 5154 ]
```

[check results](https://cran.r-project.org/web/checks/check_results_lintr.html)

> Please correct before 2024-04-01 to safely retain your package on CRAN.

### PR DESCRIPTION
**Pull request by IndrajeetPatil**
Coming from changes to the parser in R-devel

Closes #2527

### CODE REVIEW

File: tests/testthat/test-get_source_expressions.R [hunk 1]
```diff
@@ -91,10 +91,11 @@ test_that("Multi-byte characters correct columns", {
 
 test_that("Multi-byte character truncated by parser is ignored", {
   skip_if_not_utf8_locale()
+  skip_if_not_r_version("4.4.0")
```
Comment (MichaelChirico) [inline tests/testthat/test-get_source_expressions.R:94]: better would be (not fully tested)

```r
content <- "y <- x \U2013 42"
base_msg <- conditionMessage(tryCatch(str2lang(content), error = identity))
with_content_to_parse(content, {
  expect_identical(error$message, base_msg)
  ...
})
```

File: cran-comments.md [hunk 2]
```diff
@@ -8,5 +8,4 @@ This release fixes R CMD check issues in R-devel, as requested by CRAN.
 
 ## Reverse dependencies
 
-* I have run R CMD check on the downstream dependencies, no regressions were
-  found.
+* No regressions were found in R CMD check on the downstream dependencies.
```
Comment (MichaelChirico) [inline cran-comments.md:11]: revert this?

Comment (IndrajeetPatil) [reply cran-comments.md:11]: I kind of like the passive voice here because it's not clear who "I" is referring to in the active voice version. WDYT?

Comment (MichaelChirico) [reply cran-comments.md:11]: "I" refers to the person submitting to CRAN, could be "We" if we want to include all the maintainers instead.

Comment (MichaelChirico) [reply cran-comments.md:11]: Anyway, it's not clear what the change has to do with the current PR, I'd assumed this was just a "change anything" edit to get CI running.

Comment (IndrajeetPatil) [reply cran-comments.md:11]: Fair enough; I was commenting in general :) 

Let's revert this change from this PR

File: tests/testthat/test-get_source_expressions.R [hunk 3]
```diff
@@ -93,9 +93,12 @@ test_that("Multi-byte character truncated by parser is ignored", {
   skip_if_not_utf8_locale()
   # \U2013 is the Unicode character 'en dash', which is
   # almost identical to a minus sign in monospaced fonts.
-  with_content_to_parse("y <- x \U2013 42", {
-    expect_identical(error$message, "unexpected input")
-    expect_identical(error$column_number, 8L)
+  content <- "y <- x \U2013 42"
+  # message is like '<text>:1:8: unexpected invalid token\n1: ...'
```
Comment (AshesITR) [inline tests/testthat/test-get_source_expressions.R:97]: Maybe use rex with capture groups to auto-parse the base msg to expected `message` and `column`?

Comment (MichaelChirico) [reply tests/testthat/test-get_source_expressions.R:97]: I was tempted... at a certain point we're just copying the implementation straight to the test right?

how about (1) leaving literal 8 as the expected column # and (2) just checking grepl(actual, expected) to ensure we've pulled out any subset of the true error?

**Review by MichaelChirico — DISMISSED** [review]

**Review by MichaelChirico — APPROVED** [review]

**Pull request merged**